src/main/scala/io/iohk/ethereum/vm/OpCode.scala

```diff
@@ -902,7 +906,11 @@ case object SELFDESTRUCT extends OpCode(0xff, 1, 0, _.G_selfdestruct) {
     val (refund, stack1) = state.stack.pop
     val refundAddr: Address = Address(refund)
     val gasRefund: BigInt = if (state.addressesToDelete contains state.ownAddress) 0 else state.config.feeSchedule.R_selfdestruct
-    val world = state.world.transfer(state.ownAddress, refundAddr, state.ownBalance)
+
+    val world = if (state.ownAddress == refundAddr) {
+      state.world.removeEther(state.ownAddress, state.ownBalance)
+    } else
```

Comment:
Minor, style: please be consistent with the braces. Try to adhere to these guidelines: https://docs.scala-lang.org/style/